framework/tests/viewpointtest/AnnoOnTypeVariableUse.java

```diff
@@ -0,0 +1,17 @@
+import viewpointtest.quals.*;
+
```

Comment:
So who has the priority? substitution or viewpoint adaptation?
substitution replaces the type params with the concrete type arguments.
viewpoint adaptation adapts the declared type to the receiver type.

Reply:
![Screen Shot 2023-05-08 at 11 04 29 AM](https://user-images.githubusercontent.com/44760135/236859798-24a845fc-6a88-4c1f-813e-0f893522eef3.png)
there's no viewpoint adaptation for rdq generics at all?

this code has a similar error 
```
@SuppressWarnings("cast.unsafe.constructor.invocation")
class Demo<E> {
    @A E element;

    static void test() {
        Demo<@B Element> d = new @A Demo<@B Element>();
        // d.element = @A |> @RDQ = @A
        // thus expects no error here
        d.element = new @A Element();
    }
}

class Element {
    int f = 1;
}
```

Maybe the `abstractViewpointAdapter` was designed for GUT, and didn't consider annotations on typevars?

Reply:
@AndrewShf Yes, this is definitely colored by how GUT formalized viewpoint adaptation. Annotations on type variable uses was not a part of GUT, so there is no clear semantics. As discussed, please think about what a good semantics would be and how we could implement it.

Reply:
I have a design for PICO, as sent via the email. But I don't quite understand for a base type system, what kind of features we want in viewpoint adaptation generics. If allowing qualifiers on type arguments, type parameters, and field declarations, there could be multiple interpretations for a typevar use. Which one should we choose, for the general type system? I find it unclear.

Reply:
LUB?